connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/aiagent/agent/JobWorkerAgentRequestHandler.java

```diff
@@ -0,0 +1,221 @@
+import io.camunda.connector.agenticai.aiagent.memory.conversation.ConversationStoreRegistry;
+import io.camunda.connector.agenticai.aiagent.model.AgentContext;
+import io.camunda.connector.agenticai.aiagent.model.AgentResponse;
+import io.camunda.connector.agenticai.aiagent.model.JobWorkerAgentExecutionContext;
+import io.camunda.connector.agenticai.aiagent.model.JobWorkerAgentResponse;
+import io.camunda.connector.agenticai.aiagent.tool.GatewayToolHandlerRegistry;
+import io.camunda.connector.agenticai.model.message.Message;
+import io.camunda.connector.agenticai.model.tool.ToolCallProcessVariable;
+import io.camunda.spring.client.jobhandling.CommandExceptionHandlingStrategy;
+import io.camunda.spring.client.jobhandling.CommandWrapper;
+import io.camunda.spring.client.metrics.MetricsRecorder;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.CollectionUtils;
+
+public class JobWorkerAgentRequestHandler
+    extends DefaultAgentRequestHandler<JobWorkerAgentExecutionContext> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JobWorkerAgentRequestHandler.class);
+
+  private static final int MAX_ZEEBE_COMMAND_RETRIES = 3;
+
+  private final CommandExceptionHandlingStrategy defaultExceptionHandlingStrategy;
+  private final MetricsRecorder metricsRecorder;
+
+  public JobWorkerAgentRequestHandler(
+      AgentInitializer agentInitializer,
+      ConversationStoreRegistry conversationStoreRegistry,
+      AgentLimitsValidator limitsValidator,
+      AgentMessagesHandler messagesHandler,
+      GatewayToolHandlerRegistry gatewayToolHandlers,
+      AiFrameworkAdapter<?> framework,
+      AgentResponseHandler responseHandler,
+      CommandExceptionHandlingStrategy defaultExceptionHandlingStrategy,
+      MetricsRecorder metricsRecorder) {
+    super(
+        agentInitializer,
+        conversationStoreRegistry,
+        limitsValidator,
+        messagesHandler,
+        gatewayToolHandlers,
+        framework,
+        responseHandler);
+    this.defaultExceptionHandlingStrategy = defaultExceptionHandlingStrategy;
+    this.metricsRecorder = metricsRecorder;
+  }
+
+  @Override
+  protected boolean modelCallPrerequisitesFulfilled(
+      JobWorkerAgentExecutionContext executionContext,
+      AgentContext agentContext,
+      List<Message> addedUserMessages) {
+    return !CollectionUtils.isEmpty(addedUserMessages);
+  }
+
+  @Override
+  public AgentResponse completeJob(
+      JobWorkerAgentExecutionContext executionContext,
+      AgentResponse agentResponse,
+      ConversationStore conversationStore) {
+    if (agentResponse == null) {
+      LOGGER.debug(
+          "No agent response provided, completing job {} without response",
+          executionContext.job().getKey());
+      completeAsyncWithoutResponse(executionContext);
+    } else {
+      if (LOGGER.isDebugEnabled()) {
```

Comment:
:question: what's the logic behind wrapping the debug log in the `if` as you don't do it everywhere? Does it depend on what we log?

Reply:
OK, it makes sense when the optimization is worth it so that we don't _pollute_ the code with the if blocks checking whether the log level is active :+1: